### PR TITLE
JAVA-20164 Upgrade java hibernate specific modules to JDK 11

### PR DESCRIPTION
**Pull request by timis1**
Migration java related projects with hibernate

### CODE REVIEW

File: persistence-modules/java-jpa/src/test/java/com/baeldung/jpa/sqlresultsetmapping/SqlResultSetMappingUnitTest.java [hunk 1]
```diff
@@ -44,7 +44,7 @@ public void whenNamedQuery_thenSingleEntityResult() {
 
     @Test
     public void whenNamedQuery_thenMultipleEntityResult() {
-        final Query query = em.createNativeQuery("SELECT e.id, e.name, d.id, d.employeeId, d.dayOfWeek "
+        final Query query = em.createNativeQuery("SELECT e.id as idEmployee, e.name, d.id as daysId, d.employeeId, d.dayOfWeek "
```
Comment (dkapil) [inline persistence-modules/java-jpa/src/test/java/com/baeldung/jpa/sqlresultsetmapping/SqlResultSetMappingUnitTest.java:47]: why is this needed?

**Review by dkapil — CHANGES_REQUESTED** [review]

File: persistence-modules/pom.xml [hunk 2]
```diff
@@ -39,24 +39,24 @@
...
+        <module>java-jpa</module> <!-- long running -->
+        <module>java-jpa-2</module> <!-- long running -->
         <module>java-jpa-3</module>
 		<!-- enable it when persistence-modules is migrated to JDK9+ -->
 		<!-- <module>java-mongodb</module>--> <!-- long running -->
         <module>java-mongodb-2</module> <!-- long running -->
         <module>java-mongodb-3</module> <!-- long running -->
         <module>java-mongodb-queries</module> <!-- long running -->
-        <!-- <module>jnosql</module> FAILED --> <!-- long running -->
+        <module>jnosql</module> <!-- long running -->
```
[the remark this answers is not in the document]
Comment (timis1) [reply persistence-modules/pom.xml:50]: As I tested locally doesn't take to much time.

File: persistence-modules/java-jpa/src/test/java/com/baeldung/jpa/sqlresultsetmapping/SqlResultSetMappingUnitTest.java (hunk 1 above)
Comment (timis1) [reply persistence-modules/java-jpa/src/test/java/com/baeldung/jpa/sqlresultsetmapping/SqlResultSetMappingUnitTest.java:47]: It gives me an error to the column Id that is duplicated.

File: persistence-modules/java-jpa-2/src/test/java/com/baeldung/jpa/projections/HibernateProjectionsIntegrationTest.java [hunk 3]
```diff
@@ -56,16 +59,20 @@ private static Configuration getConfiguration() {
...
+
     @SuppressWarnings("deprecation")
     @Test
     public void givenProductData_whenIdAndNameProjectionUsingCriteria_thenListOfObjectArrayReturned() {
-        Criteria criteria = session.createCriteria(Product.class);
-        criteria = criteria.setProjection(Projections.projectionList()
-            .add(Projections.id())
-            .add(Projections.property("name")));
-        List<Object[]> resultList = criteria.list();
+        final CriteriaBuilder criteria = session.getCriteriaBuilder();
```
[the remark this answers is not in the document]
Comment (timis1) [reply persistence-modules/java-jpa-2/src/test/java/com/baeldung/jpa/projections/HibernateProjectionsIntegrationTest.java:67]: The `Criteria` class was removed and we can use instead `CriteriaBuilder`.

File: persistence-modules/java-jpa-2/src/main/java/com/baeldung/jpa/unrelated/entities/Cocktail.java [hunk 4]
```diff
@@ -23,22 +21,22 @@ public class Cocktail {
...
         referencedColumnName = "cocktail", 
         insertable = false, updatable = false, 
-        foreignKey = @javax.persistence
+        foreignKey = @jakarta.persistence
           .ForeignKey(value = ConstraintMode.NO_CONSTRAINT))
     private Recipe recipe;
 
     @OneToMany(fetch = FetchType.LAZY)
-    @NotFound(action = NotFoundAction.IGNORE)
+    // @NotFound(action = NotFoundAction.IGNORE)
```
[the remark this answers is not in the document]
Comment (timis1) [reply persistence-modules/java-jpa-2/src/main/java/com/baeldung/jpa/unrelated/entities/Cocktail.java:33]: The @NotFound annotation is not supported on the OneToMany.

File: persistence-modules/java-jpa-2/src/main/java/com/baeldung/jpa/text/Exam.java [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
 
-import javax.persistence.*;
+import jakarta.persistence.*;
 
 @Entity
 public class Exam {
 
     @Id
-    @GeneratedValue(strategy = GenerationType.AUTO)
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
[the remark this answers is not in the document]
Comment (timis1) [reply persistence-modules/java-jpa-2/src/main/java/com/baeldung/jpa/text/Exam.java:9]: Maybe was for testing.

File: persistence-modules/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (timis1) [reply persistence-modules/pom.xml:50]: I removed
